### PR TITLE
MGMT-12000: LVM installation failure resolved, added thinPoolConfigur…

### PR DESCRIPTION
**Pull request by Mrudraia**
…ation

## List all the issues related to this PR

- [ ] New Feature 
- [ ] Enhancement 
- [X] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [x] None

## How was this code tested?

- [X] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [] No tests needed

## Assignees

/cc @rewantsoni @avishayt 
/cc @gobindadas @gamli75 

## Checklist

- [X] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [X] Reviewers have been listed
- [X] This change does not require a documentation update (docstring, `docs`, README, etc)
- [X] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: internal/operators/lvm/manifest.go [hunk 1]
```diff
@@ -110,4 +110,8 @@ metadata:
 spec:
   storage:
     deviceClasses:
-    - name: vg1`
+    - name: vg1
+	  thinPoolConfig:
+		name: thin-pool-1
+		sizePercent: 90
+		overprovisionRatio: 10`
```
Comment (avishayt) [inline internal/operators/lvm/manifest.go:117]: Where do these values come from?

**Comment by avishayt** [comment]
Was this manually tested?  Do we have an automated test?

File: internal/operators/lvm/manifest.go (hunk 1 above)
Comment (nbalacha) [reply internal/operators/lvm/manifest.go:117]: sizePercent = 90 is the recommended default value for the field in LVMO. This states that the thinpool will use 90% of the volumegroup.
overprovisionRatio can be set to a smaller value if required.This is basically how much the storage can be overprovisioned - if the value is set to 10 and the thinpool is 10G, you can create PVCs with sizes totaling 100G .  The admin is responsible for monitoring the actual usage to ensure that all the space in the thinpool is not used up.

A smaller value may be better here. Maybe 2.

Comment (nbalacha) [reply internal/operators/lvm/manifest.go:117]: OverprovisionRatio is a multiplier, not a percentage.

Comment (akalenyu) [reply internal/operators/lvm/manifest.go:117]: gotcha, I think we can ever go lower than 2
btw -> could it be that thinPoolConfig should not have been strictly required? could users opt-out somehow?

File: internal/operators/lvm/manifests_test.go [hunk 2]
```diff
@@ -32,6 +34,7 @@ var _ = Describe("LVM manifest generation", func() {
 			}
 
 			_, err = yaml.YAMLToJSON(manifest)
+			fmt.Println(err)
```
Comment (akalenyu) [inline internal/operators/lvm/manifests_test.go:37]: nit:
you could do `Expect(err).ShouldNot(HaveOccurred(), "yamltojson err: %v", err)`

File: internal/operators/lvm/config.go [hunk 3]
```diff
@@ -12,13 +15,18 @@ type Config struct {
...
-func (o *operator) getValidDiskCount(disks []*models.Disk, installationDiskID string) int64 {
+func (o *operator) getValidDiskCount(disks []*models.Disk, installationDiskID string) (int64, error) {
 	var countDisks int64
+	var err error
 
 	for _, disk := range disks {
 		if (disk.DriveType == models.DriveTypeSSD || disk.DriveType == models.DriveTypeHDD) && installationDiskID != disk.ID && disk.SizeBytes != 0 {
-			countDisks++
+			if disk.SizeBytes < conversions.GbToBytes(o.config.LvmMinDiskSizeGB) {
+				err = fmt.Errorf("ODF LVM requires all the non-bootable disks to be more than %d GB", o.config.LvmMinDiskSizeGB)
```
Comment (akalenyu) [inline internal/operators/lvm/config.go:25]: What was the reasoning? before this PR 1 satisfying disk was enough right?

Comment (Mrudraia) [reply internal/operators/lvm/config.go:25]: The host should have atleast one non-installtion disk that can be used to provision storage. The LVM operator failure was not related to disks, It was due to yaml syntax error. As I confirmed from Storage team, we need to add thin pool configuration for LVM cluster. I copied the thin pool configuration to the LVM cluster CR, additionaly tabs were copied. Its resolved Now.

Comment (akalenyu) [reply internal/operators/lvm/config.go:25]: But here we enforce `ODF LVM requires *all* the non-bootable disks to be more than %d GB`
What am I missing?

File: internal/operators/lvm/manifest.go (hunk 1 above)
Comment (nbalacha) [reply internal/operators/lvm/manifest.go]: > gotcha, I think we can ever go lower than 2 btw -> could it be that thinPoolConfig should not have been strictly required? could users opt-out somehow?

Afraid not. The thinpoolconfig is mandatory for odf lvmo.

File: internal/operators/lvm/config.go (hunk 3 above)
Comment (akalenyu) [reply internal/operators/lvm/config.go]: awesome, thank you

**Review by avishayt — APPROVED** [review]

**Review by gamli75 — APPROVED** [review]

**Pull request merged**